### ISSUE TITLE
code cannot be run

### ISSUE THREAD

**Issue by jamesben6688** [aidialab/RetiFluidNet#1]
hi there. this code cannot be run.

In the temp.py, the output is : 

model = tf.keras.models.Model(inputs=[inputs], outputs= [outputs, output4, output3, output2, output1])

where outputs, output4, output3, output2, output1 are all with shape of (None, 256, 256, 4).

But in losses.py, in training_loss(self, y_true, y_pred) :

outputs = y_pred[:, :, :, 160:164] #Side output of the main output
output4 = y_pred[:, :, :, 164:168] #Side output4
output3 = y_pred[:, :, :, 168:172] #Side output3
output2 = y_pred[:, :, :, 172:176] #Side output2
output1 = y_pred[:, :, :, 176:180] #Side output1

which means the y_pred have at least 180 dimensions at axes -1. 

So this code cannot even compile.

Anyway to fix this?

**Comment by arminbiglari** [comment]
Hello @jamesben6688 
As our model in based on supervision techniques, we are training our model with several outputs.
the model we have presented has 4 classes. for each class we have a dice loss and bicon loss.
so what is your question? please provide more details. 
i would be happy to assist you with running the code.
